equinix/helpers_device_test.go

```diff
@@ -134,27 +127,42 @@ func Test_waitUntilReservationProvisionable(t *testing.T) {
 			args: args{
 				reservationId: "reservationId",
 				instanceId:    "instanceId",
-				meta: &packngo.Client{
-					HardwareReservations: &mockHWService{
-						GetFn: func(_ string, _ *packngo.GetOptions) (*packngo.HardwareReservation, *packngo.Response, error) {
-							return &packngo.HardwareReservation{
-								Device: nil, Provisionable: false,
-							}, nil, nil
-						},
-					},
+				handler: func(w http.ResponseWriter, r *http.Request) {
+					reservation := metalv1.HardwareReservation{
+						Device: nil, Provisionable: metalv1.PtrBool(false),
+					}
+
+					body, err := reservation.MarshalJSON()
+					if err != nil {
+						return // TODO ... ?
```

Comment:
If this code could be reached, which it seems it will not, we could return a 500 error header and the error in a {errors} body.

Reply:
Since this is a mock API, and a non-nil error would mean some kind of misconfiguration of the mock, I might lean towards panicking instead to reduce the risk that someone sees an error response from the mock and thinks it's coming from the actual API.

Reply:
Updated the tests to panic if there's an error here.  Resolving this thread but please feel free to reopen it if needed.